policykit/policyengine/views.py

```diff
@@ -824,49 +824,177 @@ def document_action_recover(request):
+        is_template=True,
+        description=desc
+    )
+    if created:
+        action_type, _ = ActionType.objects.get_or_create(codename="expensecreated")
+        policy.action_types.add(action_type)
+
+        PolicyVariable.objects.create(
+            name="yes_votes_to_approve", label="Yes Votes Needed To Approve", default_value=1, is_required=True,
+            prompt="If this number of YES votes is hit, the expense is approved", type="number", policy=policy)
+        PolicyVariable.objects.create(
+            name="no_votes_to_reject", label="No Votes Needed to Reject", default_value=1, is_required=True,
+            prompt="If this number of NO votes is hit, the expense is rejected", type="number", policy=policy)
+        PolicyVariable.objects.create(
+            name="slack_channel_id", label="Slack Channel ID", default_value="", is_required=True,
+            prompt="Which Slack Channel to use?", type="string", policy=policy)
+        
+    
+    desc = """
+    For testing: add a very simple policy so that when you post "ping" in Slack,
+    the PolicyKit app will respond "pong". Except you can customize the "pong" message!
+    """
+    policy, created = Policy.objects.get_or_create(
+        kind="trigger",
+        name="Say 'ping', Get a 'pong' Back Test Example",
+        filter='return action.text == "ping"',
+        initialize='slack.post_message(variables["pong_message"])', 
+        check='return PASSED',
+        notify='pass',
+        fail='pass',
+        is_template=True,
+        description=desc
+    )
+    if created:
+        action_type, _ = ActionType.objects.get_or_create(codename="slackpostmessage")
+        policy.action_types.add(action_type)
+
+        PolicyVariable.objects.create(
+            name="pong_message", label="What to say in response to ping", default_value="pong", is_required=True,
+            prompt="What to say in response to ping", type="string", policy=policy)
+
+    return embed_select_template(request)
+
+
+
+def get_channel_options(community_id):
+    """
+    Get channel ids and names from Slack
+    """
+    channel_options = []
+    from integrations.slack.models import SlackCommunity
+
+    slack_community = SlackCommunity.objects.get(community_id=community_id)
+    for channel in slack_community.get_conversations()['channels']:
+        if channel['is_channel']: # get only the "channels" (as opposed to group, im, mpim, private)
+            channel_options.append(
+                {'name': channel['name'], 'channel_id': channel['id']}
+            )
+    return channel_options
+
+def embed_select_template(request):
```

Comment:
with Leijie's changes, this code may need to be changed. we are planning to move away from "is_template=True" policies in general.